x-pack/plugins/saved_objects_tagging/public/management/components/table.tsx

```diff
@@ -0,0 +1,193 @@
+      'data-test-subj': 'tagsTableAction-edit',
+    });
+  }
+  if (capabilities.delete) {
+    actions.push({
+      name: i18n.translate('xpack.savedObjectsTagging.management.table.actions.delete.title', {
+        defaultMessage: 'Delete',
+      }),
+      description: i18n.translate(
+        'xpack.savedObjectsTagging.management.table.actions.delete.description',
+        {
+          defaultMessage: 'Delete this tag',
+        }
+      ),
+      type: 'icon',
+      icon: 'trash',
+      onClick: (object: TagWithRelations) => onDelete(object),
+      'data-test-subj': 'tagsTableAction-delete',
+    });
+  }
+
+  const columns = [
+    {
+      field: 'name',
+      name: i18n.translate('xpack.savedObjectsTagging.management.table.columns.name', {
+        defaultMessage: 'Name',
+      }),
+      sortable: (tag: TagWithRelations) => tag.name,
+      'data-test-subj': 'tagsTableRowName',
+      render: (name: string, tag: TagWithRelations) => {
+        return <TagBadge tag={tag} />;
+      },
+    },
+    {
+      field: 'description',
+      name: i18n.translate('xpack.savedObjectsTagging.management.table.columns.description', {
+        defaultMessage: 'Description',
+      }),
+      sortable: true,
+      'data-test-subj': 'tagsTableRowDescription',
+    },
+    {
+      field: 'relationCount',
+      name: i18n.translate('xpack.savedObjectsTagging.management.table.columns.connections', {
+        defaultMessage: 'Connections',
+      }),
+      sortable: (tag: TagWithRelations) => tag.relationCount,
+      'data-test-subj': 'tagsTableRowConnections',
+      render: (relationCount: number, tag: TagWithRelations) => {
+        if (relationCount < 1) {
+          return undefined;
+        }
+        return (
+          // eslint-disable-next-line @elastic/eui/href-or-on-click
+          <EuiLink
+            data-test-subj="tagsTableRowConnectionsLink"
+            href={getTagRelationUrl(tag)}
+            onClick={(e: React.MouseEvent) => {
+              if (!isModifiedOrPrevented(e) && e.button === 0) {
+                e.preventDefault();
+                onShowRelations(tag);
+              }
+            }}
+          >
+            <FormattedMessage
+              id="xpack.savedObjectsTagging.management.table.content.connectionCount"
+              defaultMessage="{relationCount, plural, one {1 app object} other {# app objects}}"
+              values={{ relationCount }}
+            />
+          </EuiLink>
+        );
+      },
+    },
+    ...(actions.length
+      ? [
+          {
+            name: i18n.translate('xpack.savedObjectsTagging.management.table.columns.actions', {
+              defaultMessage: 'Actions',
+            }),
+            width: '100px',
+            actions,
+          },
+        ]
+      : []),
+  ];
+
+  return (
+    <EuiInMemoryTable
+      data-test-subj="tagsManagementTable"
+      ref={tableRef as any}
+      loading={loading}
+      itemId={'id'}
+      columns={columns as any[]}
```

Comment:
Is there an EUI issue we need to address that using tables requires casting so many things?

Reply:
EUI typings around the table components are quite a pain when you have mixed attribute/action columns, due to the fact most types are complex and/or hybrid. TS get lost very easily, and it requires to force/explicit-cast a lot of things along the way if we want to avoid this final `as any[]`. Can try again though.

Reply:
edit: resolved